### DISCUSSION TITLE
Using raw cells to define `\newcommand` in LaTeX

### DISCUSSION THREAD

**Discussion by binary10** [sagemathinc/cocalc#6209] [Q&A]
In a Jupyter notebook, I created a raw cell with a `\newcommand` definition for LaTeX. I'm running into two issues:

1. The definition does not take effect in the notebook when I try to use it in another markdown cell. However, if I export the notebook with `nbconvert` via the menu, the document renders the command. Is there a way for the notebook to use definitions in raw cells?
2. I want to reuse some definitions repeatedly **across** notebooks. Is there a way to store these commands in a file so that new notebooks will have access to them?

[This discussion](https://github.com/sagemathinc/cocalc/discussions/6177) relates to creating new commands, however it only refers to defining new commands that can be used across multiple cells in a single document. My question relates to reusing commands across many documents.

**Comment by mforbes** [comment]
@binary10 I tried to solve 2 by a dynamic import that defines a bunch of macros: I run this at the top of ever notebook

```
import mmf_setup;mmf_setup.nbinit()
```

This also allows me to dynamically update the path (I use a flag in `pyproject.toml`) and some other things.  Somehow this is broken on CoCalc right now (https://github.com/sagemathinc/cocalc/issues/6210) but this approach works across other Jupyter notebook environments.  We have discussed other solutions such as providing a set of macros across `.md` files too, but this would probably be CoCalc specific.

**Reply by binary10** [reply]
I go between Python and R, so this is only a partial workaround for my uses. I'll keep looking for a more general solution, but thanks for responding.
